azure-pipelines.perf-test.yml

```diff
@@ -21,31 +21,49 @@ steps:
       filePath: yarn-ci.sh
     displayName: yarn
 
-  # @fluentui/perf-test-northstar needs build and bundle steps
   - script: |
-      yarn lage build --to @fluentui/perf-test-northstar --to perf-test --verbose
-    displayName: build to perf-test
+      yarn perf:test:affected
+    workingDirectory: packages/fluentui/perf-test-northstar
+    displayName: Check if @fluentui/react-northstar was modified
 
-  # Fluent perf-test must run before Fabric perf-test until they are consolidated.
-  # @fluentui/perf-test-northstar Baseline
   - script: |
-      yarn perf:test:base
-    condition: eq(variables.isPR, false)
-    workingDirectory: packages/fluentui/perf-test-northstar
-    displayName: Perf Test Base (Fluent)
+      yarn perf:test:affected
+    workingDirectory: apps/perf-test
+    displayName: Check if @fluentui/react was modified
+
+  # Build @fluentui/perf-test-northstar
+  - script: |
+      yarn lage build --to @fluentui/perf-test-northstar --to @fluentui/react-conformance --verbose
+    condition: and(eq(variables['RunNorthstarPerfTest'], true), eq(variables.isPR, true))
+    displayName: Build to Perf Test (Northstar)
+
+  # Build Fabric Perf Test
+  - script: |
+      yarn lage build --to perf-test --verbose
+    condition: and(eq(variables['RunPerfTest'], true), eq(variables.isPR, true))
+    workingDirectory: apps/perf-test
+    displayName: Build to Perf Test
 
-  # @fluentui/perf-test-northstar PR
+  # Run @fluentui/perf-test-northstar
   - script: |
       yarn perf:test
-    condition: eq(variables.isPR, true)
+    condition: and(eq(variables['RunNorthstarPerfTest'], true), eq(variables.isPR, true))
```

Comment:
I wonder if these conditions and variables are maintainable/scalable. Might be worth just calling the ADO API to cancel a job ?

Reply:
Not familiar enough with the ADO API right now but I’m curious to know how you think that would be used here? 😮 Seemed to me that using a condition to determine if a step should be run or skipped is more straightforward 🤔

Reply:
Well now we have an extra env variable to be aware of, and it's not too obvious that we should be aware of it, and in the future we might now know easily where this env variable would even be used.

Creating a task that actually programmatically cancels the job, stops there, no other tasks or potentially code in our repo needs to know that it was even affected at all

Reply:
So after reading through the ADO API docs, there isn’t actually a way to just cancel a job by itself. The closest thing to that is cancelling the whole build which isnt ideal since that also cancels the other jobs running in the that build. There’s also a delay on when the build actually gets cancelled as well which takes up extra time and resources. Just using a task condition seems to be the fastest and cleanest way to go about this issue since it quickly just skips a task. Since I separated the perf test pipeline to two jobs, I switched to just using one variable called `PackageAffected` and that would be the only variable (along with `isPR`) needed for all of the perf test pipeline task conditions. That should simplify it a bit as making a search for that keyword in our repo should point that person to the two (currently) perf packages along with the perf pipeline yaml file.

Reply:
You could try writing a quick bash version of this script: https://stackoverflow.com/questions/62044055/is-it-possible-to-cancel-a-azure-devops-pipeline-job-programmatically

the access token seems to already exist  as a part of the build env variables.

Reply:
well that really sucks 😓

Reply:
well I can't see any other way, @ecraig12345  do you have any better ideas ?

Reply:
If we can't think of anything else, maybe move the "what should we run" checks to a separate job and make the others conditional on that? Save the check results in output variables, then make each subsequent job dependent on the checks job and add a condition checking the checks job's output variables (assuming those are available in the context of conditions). @TristanWatanabe let me know if that didn't make sense and I can try to explain it.

https://stackoverflow.com/questions/57485621/share-variables-across-stages-in-azure-devops-pipelines (references stages but it looks like it should be similar for jobs)
https://gaunacode.com/passing-variables-between-jobs-for-azure-devops-pipelines
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml